src/frontend/src/pages/TeamsPage/DescriptionPageBlock.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Button, IconButton, TextField, useTheme } from '@mui/material';
+import { useState } from 'react';
+import { useAuth } from '../../hooks/auth.hooks';
+import { useEditTeamDescription } from '../../hooks/teams.hooks';
+import { Team } from 'shared';
+import { Edit } from '@mui/icons-material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import PageBlock from '../../layouts/PageBlock';
+import ReactMarkdown from 'react-markdown';
+import styles from '../../stylesheets/pages/teams.module.css';
+
+interface DescriptionPageBlockProps {
+  team: Team;
+}
+
+const DescriptionPageBlock: React.FC<DescriptionPageBlockProps> = ({ team }) => {
+  const auth = useAuth();
+  const theme = useTheme();
+  const [isEditingDescription, setIsEditingDescription] = useState(false);
+  const [description, setDescription] = useState(team.description);
+  const [isPreview, setIsPreview] = useState(false);
+  const { isLoading, isError, error, mutateAsync } = useEditTeamDescription(team.teamId);
+
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (isLoading) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    if (!isUnderWordCount(description, 300)) {
+      return alert('Description must be less than 300 words');
+    }
+    await mutateAsync(description);
+    setIsEditingDescription(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+
+  const countWords = (str: string): number => {
```

Comment:
if we're counting words here and in thebackend we can put the function in shared instead of dpulicating code